builder/build_env.py

```diff
@@ -405,23 +410,27 @@ def build_env(arg_strings=None):
             return result
 
         # Add dependency tree information to the packages list
-        dep_tree = _create_dep_tree(recipes)
+        dep_tree += _create_dep_tree(variant_recipes, len(recipes))
+        recipes += variant_recipes
 
-        # Build each package in the packages list
-        for recipe in _traverse_recipes(recipes, dep_tree):
-            package_build_args = ["--working_directory", recipe['repository']]
+    # Build each package in the packages list
+    for recipe in _traverse_recipes(recipes, dep_tree):
+        package_build_args = ["--working_directory", recipe['repository']]
 
-            for channel in recipe.get('channels', []):
-                package_build_args += ["--channels", channel]
+        for channel in recipe.get('channels', []):
+            package_build_args += ["--channels", channel]
 
-            if 'recipe' in recipe:
-                package_build_args += ["--recipes", recipe['recipe']]
+        package_build_args += ["--python_versions", recipe['python']]
```

Comment:
@bethune-bryant I had to move these before adding the `--recipes` option. It looks line some of the tests rely on `--recipes` being the last argument passed to `build_feedstock`.

Reply:
Yeah, it looks like `validate_build_feedstock` is grabbing the last argument as the package being built.